src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,99 @@
+package duke;
+
+import java.util.Scanner;
+
+import duke.command.AddCommand;
+import duke.command.CommandType;
+import duke.command.DeleteCommand;
+import duke.command.EmptyCommand;
+import duke.command.ExitCommand;
+import duke.command.FindCommand;
+import duke.command.ICommand;
+import duke.command.ListCommand;
+import duke.command.MarkCommand;
+import duke.command.SaveCommand;
+import duke.command.UnmarkCommand;
+
+/**
+ * Represents a parser to parse string inputs from user
+ * and output appropriate commands to be executed.
+ */
+public class Parser {
+    /**
+     * Returns appropriate command from parsed input.
+     * @param input
+     * @return Command from parsed input to be executed.
+     */
+    public static ICommand parse(String input) {
+        Scanner sc = new Scanner(input);
+        try {
+            String command = sc.next();
+            if (!CommandType.COMMAND_MAP.containsKey(command)) {
```

Comment:
This conditional is a little hard to understand. Perhaps abstracting a method to make it more readable?